### PR TITLE
Add options to disable the audio system for audio recordings

### PR DESCRIPTION
**Pull request by rolandlo**
For one session the audio system can be disabled from the command line using the --disable-audio flag This option is particularly useful for troubleshooting crashes on startup. For future sessions the audio system can be disabled from the preferences by enabling the "Disable audio" checkbox.

### CODE REVIEW

**Comment by rolandlo** [comment]
Fixes #3905. Fixes #3531. Partly fixes #5020,

File: src/core/control/Control.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class Control:
         public ClipboardListener,
         public ProgressListener {
 public:
-    Control(GApplication* gtkApp, GladeSearchpath* gladeSearchPath);
+    Control(GApplication* gtkApp, GladeSearchpath* gladeSearchPath, gboolean disableAudio);
```
Comment (bhennion) [inline src/core/control/Control.h:78]:
```suggestion
    Control(GApplication* gtkApp, GladeSearchpath* gladeSearchPath, bool disableAudio);
```
convertion from gboolean to bool is safe (and is done anyway implicitely in your `? :` construct)

**Review by bhennion — CHANGES_REQUESTED** [review]
The code looks good to me. I think the toolbar items (record, pause/play, forward, backwards, stop and the "Play object" tool) and menu entries should be disabled as well though (or even removed). Right now, if the audio is disabled and we click to record, it all looks like it's recording (but it's really not, ofc). This is confusing.

File: src/core/control/Control.h (hunk 1 above)
Comment (rolandlo) [reply src/core/control/Control.h]: done

**Comment by rolandlo** [comment]
> The code looks good to me. I think the toolbar items (record, pause/play, forward, backwards, stop and the "Play object" tool) and menu entries should be disabled as well though (or even removed). Right now, if the audio is disabled and we click to record, it all looks like it's recording (but it's really not, ofc). This is confusing.

Thanks for the review! The audio GUI elements are now hidden when audio is disabled. I think hiding them is better than disabling them, since they will only disturb.

**Review by bhennion — APPROVED** [review]
LGTM (up to the following nitpick)

File: src/core/control/Control.cpp [hunk 2]
```diff
@@ -1001,6 +1002,9 @@ void Control::actionPerformed(ActionType type, ActionGroup group, GtkToolButton*
 
         case ACTION_AUDIO_RECORD: {
             bool result = false;
+            if (!audioController) {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (bhennion) [reply src/core/control/Control.cpp:1007]: Actually, Plugins may still trigger those... Maybe just add a g_warning in the if statements

**Comment by rolandlo** [comment]
Merging in 24 hours if no objections are raised.

File: src/core/control/Control.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolandlo) [reply src/core/control/Control.cpp]: I have added the `g_warning`s.

**Pull request merged**